cyral/utils/testutils.go

```diff
@@ -266,3 +266,100 @@ func FormatDatadogIntegrationDataIntoConfig(name, apiKey string) string {
 		api_key = "%s"
 	}`, name, apiKey)
 }
+
+// FormatPolicyIntoConfig formats a policy map into a Terraform configuration string.
+//
+// resName is the resource name to be used in the Terraform configuration.
+// policy is a map containing the policy data.
+func FormatPolicyIntoConfig(resName string, policy map[string]interface{}) string {
+	var config strings.Builder
+
+	config.WriteString(fmt.Sprintf(`
+resource "cyral_policy_v2" "%s" {
+`, resName))
+
+	if name, ok := policy["name"]; ok {
+		config.WriteString(fmt.Sprintf("  name        = \"%s\"\n", name))
+	}
+	if description, ok := policy["description"]; ok {
+		config.WriteString(fmt.Sprintf("  description = \"%s\"\n", description))
+	}
+	if enabled, ok := policy["enabled"]; ok {
+		config.WriteString(fmt.Sprintf("  enabled     = %v\n", enabled))
+	}
+	if tags, ok := policy["tags"]; ok {
+		config.WriteString(fmt.Sprintf("  tags        = %s\n", FormatTags(tags)))
+	}
+	if scope, ok := policy["scope"]; ok {
+		config.WriteString(fmt.Sprintf("  scope       %s\n", FormatScope(scope)))
+	}
+	if validFrom, ok := policy["valid_from"]; ok {
+		config.WriteString(fmt.Sprintf("  valid_from  = \"%s\"\n", validFrom))
+	}
+	if validUntil, ok := policy["valid_until"]; ok {
+		config.WriteString(fmt.Sprintf("  valid_until = \"%s\"\n", validUntil))
+	}
+	if document, ok := policy["document"]; ok {
+		escapedDocument := strings.ReplaceAll(document.(string), "${", "$${")
+		config.WriteString(fmt.Sprintf("  document    = chomp(\n<<EOT\n%s\nEOT\n)\n", escapedDocument))
+	}
+	if enforced, ok := policy["enforced"]; ok {
+		config.WriteString(fmt.Sprintf("  enforced    = %v\n", enforced))
+	}
+	if policyType, ok := policy["type"]; ok {
+		config.WriteString(fmt.Sprintf("  type        = \"%s\"\n", policyType))
+	}
+
+	config.WriteString("}\n")
+
+	return config.String()
+}
+
+// FormatScope formats the scope data into a string suitable for Terraform configuration.
+//
+// scope is an interface that should be a map with string keys and slice of strings as values.
+func FormatScope(scope interface{}) string {
```

Comment:
I understand that `FormatPolicyIntoConfig` was put here so it can be reused for other tests, but it doesn't seem to be the case that we would reuse `FormatScope` or `FormatTags`, correct? If I'm correct, they could be changed to `formatScope` and `formatTags`/

Reply:
fixed